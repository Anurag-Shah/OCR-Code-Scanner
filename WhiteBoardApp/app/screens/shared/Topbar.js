import React from 'react';
import { Text, View, StyleSheet, TouchableOpacity, Platform } from 'react-native';
import { DrawerActions } from '@react-navigation/native';
import { Ionicons } from '@expo/vector-icons';

export default class Topbar extends React.Component {
  render() {
    return (
      <View style={[styles.container, styles.statusBarMargin]} >
        <TouchableOpacity onPress={() => { this.props.navigation.dispatch(DrawerActions.openDrawer()) }}>
          <Ionicons name="ios-list" size={32} style={{ color: 'black' }} />
        </TouchableOpacity>

        <Text style={styles.title}>{this.props.title}</Text>
        {this.props.title === 'Camera' ? (
          <TouchableOpacity onPress={() => this.props.navigation.navigate('TextEditorPage')}>
            <Ionicons name="clipboard-outline" size={32} style={{ color: 'black' }} />
          </TouchableOpacity>
        ) : (
          <TouchableOpacity onPress={() => this.props.navigation.navigate('Camera')}>
            <Ionicons name="camera-outline" size={32} style={{ color: 'black' }} />
          </TouchableOpacity>
        )

        }
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    width: "100%",
    height: (Platform.OS === 'ios') ? 44 : 60,
    flexDirection: "row",
    justifyContent: "space-between",
    alignItems: "center",
    paddingHorizontal: 20,
    backgroundColor: 'white'
  },
  statusBarMargin: {
    marginTop: (Platform.OS === 'ios') ? 0 : 24,
  },
  title: {
    fontSize: 24,
    fontWeight: 'bold',
    color: '#336600'
  },
});