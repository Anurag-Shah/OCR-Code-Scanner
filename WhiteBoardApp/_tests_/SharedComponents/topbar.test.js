import React from 'react';
import 'react-native';
import renderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react-native';


import Topbar from '../../app/screens/shared/Topbar';

describe('<Topbar />', () => {
    // expect(test.someFunc(val)).toEqual(someVal);
    it('renders correctly', () => {
        let topbar = renderer.create(<Topbar />);
        const tree = topbar.toJSON();
        expect(tree).toMatchSnapshot();
    });

    describe('Test Title', () => {
        it('Hi', () => {
            const { getByText } = render(<Topbar title='Hi' />);
            const title = getByText('Hi');
            expect(title).toBeTruthy();
        });

        it('Camera', () => {
            let topbar = renderer.create(<Topbar title='Camera' />).getInstance();
            expect(topbar.props.title).toEqual('Camera');
        });
        it('Team18', () => {
            let topbar = renderer.create(<Topbar title='Team18' />).getInstance();
            expect(topbar.props.title).toEqual('Team18');
        });
    });

});