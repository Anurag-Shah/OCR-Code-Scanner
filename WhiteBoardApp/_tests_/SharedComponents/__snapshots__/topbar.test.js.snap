// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Topbar /> renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "flexDirection": "row",
        "height": 44,
        "justifyContent": "space-between",
        "paddingHorizontal": 20,
        "width": "100%",
      },
      Object {
        "marginTop": 0,
      },
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text />
  </View>
  <Text
    style={
      Object {
        "color": "#336600",
        "fontSize": 24,
        "fontWeight": "bold",
      }
    }
  >
    WhiteBoard
  </Text>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text />
  </View>
</View>
`;
